rust/tool/src/install.rs

```diff
@@ -163,24 +163,34 @@ impl Installer {
     where
         F: FnMut(&mut Self, &Generation, &mut GenerationArtifacts) -> Result<()>,
     {
-        for link in links {
-            let generation_result = Generation::from_link(link)
-                .with_context(|| format!("Failed to build generation from link: {link:?}"));
-
-            // Ignore failing to read a generation so that old malformed generations do not stop
-            // lzbt from working.
-            let generation = match generation_result {
-                Ok(generation) => generation,
-                Err(e) => {
-                    log::debug!(
-                        "Ignoring generation {} because it's malformed.",
-                        link.version
-                    );
-                    log::debug!("{e:#}");
-                    continue;
+        let generations = links
+            .iter()
```

Comment:
`into_iter` could be used here right?
otherwise `iter` will collect `&&GenerationLink` I assume?

Reply:
We collect `Vec<Generation>`, but i can make it more obvious by adding the type.